torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -6943,10 +6943,65 @@ def test_ddp_multiple_nested_unused_params_err_ignore_params(self):
             # certain parameters.
             self._test_ddp_multiple_nested_unused_params_error(ignore_sparse=True)
 
-        @unittest.skipIf(
-            BACKEND != "nccl" and BACKEND != "gloo",
-            "Only Nccl & Gloo backend support DistributedDataParallel",
-        )
+        @unittest.skipIf(BACKEND != 'nccl' and BACKEND != 'gloo',
+                         "Only Nccl & Gloo backend support DistributedDataParallel")
+        @skip_if_lt_x_gpu(2)
+        def test_ddp_inference(self):
+            # tests that DDP module can be run on a single node with no_grad
+            # or eval setting and there is no hang.
+            rank = self.rank
+            torch.cuda.set_device(rank)
+            model = Net().cuda()
+            local_model = copy.deepcopy(model)
+            model = torch.nn.parallel.DistributedDataParallel(
+                model,
+                device_ids=[rank],
+            )
+            inp = torch.randn(10, 2, device=rank)
+            if self.rank == 0:
+                with torch.no_grad():
+                    for _ in range(6):
+                        ddp_out = model(inp)
+                        local_out = local_model(inp)
+                        self.assertEqual(ddp_out, local_out)
+                    torch.cuda.synchronize()
+
+                model.eval()
+                for _ in range(6):
+                    ddp_out = model(inp)
+                    local_out = local_model(inp)
+                    self.assertEqual(ddp_out, local_out)
+                torch.cuda.synchronize()
+
+            self._barrier(timeout=30)
```

Comment:
Are all the 3 barriers here necessary, even after `cuda.sync`? Why do they need a non-default higher timeout here?

Reply:
I don't think the calls to `synchronize` are necessary actually, will remove those.

The calls to barrier are there because we're only running inference on rank 0, if the inference unexpectedly takes too log, the default barrier which I believe has a timeout of 10s can timeout, leading to a false positive failure.

In this test the inference does not take nearly the full 30s but wanted to have plenty of buffer to avoid flakiness.